src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -252,7 +252,7 @@ public Packaging.PackageBuilder CreateBuilder(string basePath, NuGetVersion vers
             Packaging.Manifest manifest = null;
 
             // If there is a project.json file, load that and skip any nuspec that may exist
-            if (!PackCommandRunner.ProcessProjectJsonFile(builder, basePath, builder.Id, version, suffix, GetPropertyValue))
+            if (!PackCommandRunner.ProcessProjectJsonFile(builder, _project.DirectoryPath as string, builder.Id, version, suffix, GetPropertyValue))
```

Comment:
_project is dynamic, and if i don't cast it complains that method group or anonymous function cannot be used as a constituent value of a dynamically bound operation.